lib/OpenQA/WebAPI/Controller/Session.pm

```diff
@@ -76,10 +48,7 @@ sub destroy {
     my $auth_method = $self->app->config->{auth}->{method};
     my $auth_module = "OpenQA::WebAPI::Auth::$auth_method";
     eval { $auth_module->import('auth_logout'); };
-    if (!$@) {
-        auth_logout($self);
-    }
-
+    auth_logout($self) unless $@;
```

Comment:
Importing in the middle of code that runs over and over is rather ugly. How about something like this?
```perl
if (my $sub = $auth_module->can('auth_logout')) { $self->$sub) }
```

Reply:
done